### ISSUE TITLE
Extend TTNN Documentation with Tutorials and Best Practices for Model Conversion

### ISSUE THREAD

**Issue by npenkov-epam** [tenstorrent/tt-metal#35854]
I had great experience working with the examples in the demo to create a model conversion. However finding and understanding understanding  the code examples took me some time. I think that it could be beneficial for new users if the current documentation provides more tutorials and references on this topic.

To enhance the developer experience, I suggest:

- Adding dedicated documentation pages with step-by-step tutorials for model conversion
- Including best practices for model conversion (e.g. which ttnn tile layout corresponds to which standard torch block)
- Providing detailed explanations of specific SDK and API functionality related to model conversion (e.g. usage of [attention_mask_functions.py](https://github.com/tenstorrent/tt-metal/blob/main/models/experimental/functional_common/attention_mask_functions.py)).

These improvements would make it easier for developers that are getting started to use the tooling more effectively.

**Comment by npenkov-epam** [comment]
Please add label EPAM_DX

**Comment by aczajkowskiTT** [comment]
@pbaraTT What we want to do here, is include this in the golden paths for the critical OPs in models (the table you did recently)
